holoviews/plotting/bokeh/element.py

```diff
@@ -434,6 +445,20 @@ def _axis_properties(self, axis, key, plot, dimension=None,
                 if jsfunc:
                     formatter = FuncTickFormatter(code=jsfunc)
                     axis_props['formatter'] = formatter
+
+        if axis == 'x':
```

Comment:
If you use the `invert_axes` then the y-axis can be the categorical axis. Additionally the arguments already provide you with the desired axis, so I think this is sufficient:

```python
if isinstance(axis, CategoricalAxis):
    for key in list(axis_props):
        if key.startswith('major_label'):
            new_key = key.replace('major_label', 'group')
            axis_props[new_key] = axis_props[key]
```

You may also have to pop the old key from ``axis_props``, but since you weren't doing that maybe not.

Reply:
Okay, but I believe I will still get the error:

```
/mnt/c/Users/user/GOOGLE~1/Bash/bokeh/bokeh/core/has_props.py in apply_theme(self, property_values)
    589         old_values = dict()
    590         for k in added.union(removed):
--> 591             old_values[k] = getattr(self, k)
    592 
    593         if len(property_values) > 0:

AttributeError: 'LinearAxis' object has no attribute 'group_text_font_style'
```

Reply:
Also axis in this case is a string 'x' or 'y' I think...